src/index.js

```diff
@@ -263,6 +269,17 @@ export default class extends Component {
     if (!this.state.offset || width !== this.state.width || height !== this.state.height) {
       state.offset = offset
     }
+
+    // related to https://github.com/leecade/react-native-swiper/issues/570
+    // contentOffset is not working in react 0.48.x so we need to use scrollTo
+    // to emulate offset.
+    if (Platform.OS === 'ios') {
+      if (this.initialRender && this.props.loop && this.state.total > 1) {
```

Comment:
Why only if loop is enabled?

Reply:
It does not need to make any offset if the loop is not enabled.
In case if the loop is enabled - it unshifting the last page in front of the pages array, and pushing the first page to the back of the array - in this case, we need to maintain offset.